ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -123,31 +134,26 @@ class JITModBuilder {
       }
       // If this is a recursive call that has been inlined, remove the
       // inlined code and turn it into a normal call.
-      for (auto Tup : InlinedCalls) {
-        CallInst *CInst = get<1>(Tup);
-        // Have we inlined this call already? Then this is recursion.
-        if (CInst->getCalledFunction() == CF) {
-          if (VMap[CF] == nullptr) {
-            StringRef CFName = CF->getName();
-            // Declare function.
-            auto DeclFunc = llvm::Function::Create(CF->getFunctionType(),
-                                                   GlobalValue::ExternalLinkage,
-                                                   CFName, JITMod);
-            VMap[CF] = DeclFunc;
-            for (size_t i = 0; i < FAddrLen; i++) {
-              char *FName = FAddrKeys[i];
-              uint64_t FAddr = FAddrVals[i];
-              if (strcmp(FName, CFName.data()) == 0) {
-                globalMappings.insert(pair<StringRef, uint64_t>(CFName, FAddr));
-                break;
-              }
+      if (isRecursiveCall(CF)) {
+        if (VMap[CF] == nullptr) {
+          StringRef CFName = CF->getName();
+          // Declare function.
+          auto DeclFunc = llvm::Function::Create(CF->getFunctionType(),
+                                                 GlobalValue::ExternalLinkage,
+                                                 CFName, JITMod);
+          VMap[CF] = DeclFunc;
+          for (size_t i = 0; i < FAddrLen; i++) {
+            char *FName = FAddrKeys[i];
+            uint64_t FAddr = FAddrVals[i];
```

Comment:
Is it worth breaking this out as a separate variable? It only seems to be used once (on line 149)?

Reply:
I find it improves readability. But I don't feel strongly about it. `FName` is also only used once.

Reply:
I don't feel strongly either, so do as you see fit.

Reply:
I think I prefer it as is.